include/cucumber-cpp/internal/step/BreakpointSupport.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef CUKE_BREAKPOINTSUPPORT_HPP_
+#define CUKE_BREAKPOINTSUPPORT_HPP_
+
+#include "../CukeDll.hpp"
+
+namespace cucumber {
+namespace internal {
+
+#if defined(_MSC_VER)
+// The ", true" makes the expression to be of type bool, which is required
+// for the CUKE_BREAK_INTO_DEBUGGER macro
+#define CUKE_BREAK_INTO_DEBUGGER (__debugbreak(), true)
```

Comment:
The `CUKE_BREAK_INTO_DEBUGGER` implementation comes from ABL's assert implementation, where it was called in an expression context.  `__debugbreak()` though has a type of `void`, which leads to compiler errors.

The same applies to the clang branch below.  `asm` is a statement and can't be called in an expression context.

It seems though that breaking into the debugger here doesn't require the macro to behave as an expression.